Feature: Add generic Iframe support

Blendman974 opened the pull request:
Add the command /iframe.
![outline-iframe](https://github.com/user-attachments/assets/d4497e7f-1138-46de-88eb-b4d53c0b2325)

Simpler version of #6947 :  
Added a generic `EmbedDescriptor` that matches all links.  

To prevent making all links Iframes (when pasting for example), there is a  `noAutoMatch` boolean (default to false) on the `EmbedDescriptor` that prevents it from being processed as an embed on paste.

Icon is from #5137

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/outline/outline?pullRequest=7319) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/outline/outline?pullRequest=7319) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/outline/outline?pullRequest=7319) it.</sub>

tommoor commented on shared/editor/embeds/index.tsx:
```diff
@@ -621,6 +624,17 @@ const embeds: EmbedDescriptor[] = [
   }),
+  /* The generic iframe embed should always be the last one */
+  new EmbedDescriptor({
+    title: "Iframe",
```
```suggestion
    title: "Embed",
```

This seems like better language for the average user, iframe can remain as a search keyword.

tommoor commented on shared/editor/embeds/index.tsx:
```diff
@@ -55,6 +55,8 @@ export class EmbedDescriptor {
   /** Whether the bottom toolbar should be hidden – use this when the embed itself includes a footer */
   hideToolbar?: boolean;
+  /** Whether the embed should be auto matched when pasting a URL */
+  noAutoMatch?: boolean;
```
```suggestion
  matchOnPaste?: boolean;
```

Maybe we just get explicit with this?

Blendman974 commented:
Renamed "Iframe" to "Embed" (also changed the asset name for consistency)
And changed the boolean to matchOnPaste. (defaults to true so we don't need to add it on every existing Embeds)
Thx @tommoor

rcronin commented:
Any idea when this will get merged into the code base? Really excited about this feature

tommoor commented:
Needs testing and probably will choose another image from our existing libraries, but other than that it's likely to be in v0.79.0

airbenich commented:
How is the size (width & height) of the embeding calculated? Can the width & height be set manually?

Geequlim commented:
This feature is simply amazing! Please merge this PR as soon as possible and release a new version—I can't wait any longer!

florian-obradovic commented:
Awesome - I get a refused to connect:
![CleanShot 2024-09-05 at 21 07 06@2x](https://github.com/user-attachments/assets/7f756083-e0f7-48a8-9ca1-73987f4f4671)

tommoor commented:
In that case Plane is blocking the frame from rendering, there's nothing we can do on Outline side.

florian-obradovic commented:
@tommoor thanks.
Any suggestions what causes the block?
So, is it a web server setting or in the application code?
It happens with all sites I tried.

tommoor commented:
If you open developer tools in the browser, the error will be visible in the network tab.
